src/Sarif.UnitTests/TestData/PrereleaseCompatibilityTransformer/ExpectedOutputs/ComprehensiveFileProperties.sarif

```diff
@@ -29,7 +29,7 @@
       ],
       "results": [
         {
-          "ruleId": "C1",
+          "ruleIndex": 0,
```

Comment:
>ruleIndex [](start = 11, length = 9)

I don't think this is the desired outcome. I think you should always include `ruleId` for readability. As your "rule id collisions" test demonstrates, you do mention both `ruleId` and `ruleIndex` when there is a collision. I think you should do it even if there is no collision.